varanus/src/main/java/com/yelp/android/varanus/NetworkMonitor.kt

```diff
@@ -24,7 +25,7 @@ class NetworkMonitor(
 ) {
     private val networkTrafficAlerter =
             LogUploadingManager(alertIssuer, windowLength, clear_increment)
-    private var endpoints = HashMap<String, EndpointSpecificNetworkTracker>()
+    private var endpoints = ConcurrentHashMap<String, EndpointSpecificNetworkTracker>()
```

Comment:
Since ConcurrentHashMap works across threads, simply checking for a key being present isn't enough to guarantee that you are not overwriting a value in the map. There a couple mechanisms built in its api that simplify this operation.

So addLog should be modified to look something like:
```kotlin
val endpointTracker = endpoints[log.endpoint] ?: endpoints.computeIfAbsent {
    EndpointSpecificNetworkTracker(log.endpoint, windowLength, persister, networkTrafficAlerter)
}
endpointTracker.addLogAndPersist(log)
networkTrafficAlerter.registerLogs(endpoints)
```

Also the parameters of this class appear to have different indentation.

Reply:
We don't have computeIfAbsent for the current min api, so I used putIfAbsent instead.  I'll leave this open till tomorrow in case you have any opinions on that